bentoml/frameworks/transformers.py

```diff
@@ -0,0 +1,164 @@
+import os
+from bentoml.service import BentoServiceArtifact
+from bentoml.exceptions import MissingDependencyException, InvalidArgument, NotFound
+
+try:
+    import transformers
+except ImportError:
+    raise MissingDependencyException(
+        "transformers package is required to use TransformersModelArtifact"
+    )
+
+
+class TransformersModelArtifact(BentoServiceArtifact):
+    """[summary]
+    Abstraction for saving/loading HuggingFace Transformers models
+
+    Args:
+        name (string): name of the artifact
+
+    Raises:
+        MissingDependencyException: transformers package is required for TransformersModelArtifact
+        InvalidArgument: invalid argument type, model being packed 
+            must be either a dictionary of format 
+            {'model':transformers model object,'tokenizer':transformers tokenizer object} 
+            or a directory path where the model is saved 
+            or a pretrained model provied by transformers 
+            which can be loaded by transformers.AutoModel
+        NotFound: if the provided model name or model path is not found
+
+    Example usage:
+
+    >>> 
+    >>> import bentoml
+    >>> from bentoml.adapters import JsonInput
+    >>> from bentoml.frameworks.transformers import TransformersModelArtifact
+    >>>
+    >>> # Example : text generation using GPT2
+    >>> # Explicitly add either torch or tensorflow dependency which will be used by transformers
+    >>> @bentoml.env(pip_dependencies=["torch==1.6.0", "transformers==3.1.0"])
+    >>> @bentoml.artifacts([TransformersModelArtifact('gptModel')])
+    >>> class TransformersService(bentoml.BentoService):
+    >>>     @bentoml.api(input=JsonInput())
+    >>>     def predict(self, parsed_json):
+    >>>          src_text = parsed_json[0].get("text")
```

Comment:
No need to change anything here, just a quick note:

This implementation works when micro-batch is disabled, but will only process the first item when micro-batching is enabled. A better way to process this would be using the library's batch processing API to process a batch of input data at once. Or as a step gap, to use a for loop to process each item in the array. 

I'm thinking about allowing the user to create inference API that does not support micro-batching and simply their code, so use cases like this could be simpler.  e.g. with just a single item when micro batching is not needed `src_text = parsed_json.get("text")`.

Reply:
Thanks a lot for the suggestion. I was trying to do something similar since our use case requires micro batching to be disabled and accepts only a single input. This was just a simple workaround that I quickly put in.